lib/chef/resource/runit_service.rb

```diff
@@ -0,0 +1,693 @@
+
+      property :restart_on_update, [TrueClass, FalseClass], default: true,
+               description: "Whether the service should be restarted when the run script is updated. Defaults to true. Set to false if the service shouldn't be restarted when the run script is updated."
+
+      property :run_template_name, String, default: lazy { service_name },
+               description: "An alternate filename of the run run script to use instead of the value provided in service_name."
+
+      property :log_template_name, String, default: lazy { service_name },
+               description: "An alternate filename of the log run script to use instead of the value provided in service_name."
+
+      property :check_script_template_name, String, default: lazy { service_name },
+               description: "An alternate filename for the check script to use instead of the value provided in service_name."
+
+      property :finish_script_template_name, String, default: lazy { service_name },
+               description: "An alternate filename for the finish script to use instead of the value provided in service_name."
+
+      property :control_template_names, Hash, default: lazy { set_control_template_names },
+               description: "A hash of control signals (see control above) and their alternate template name(s) replacing service_name."
+
+      property :status_command, String, default: lazy { "#{sv_bin} status #{service_name}" },
+               description: "The command used to check the status of the service to see if it is enabled/running (if it's running, it's enabled). This hardcodes the location of the sv program to /usr/bin/sv due to the aforementioned cookbook load order."
+
+      property :sv_templates, [TrueClass, FalseClass], default: true, description: "If true, the :enable action will create the service directory with the appropriate templates. Default is true. Set this to false if the service has a package that provides its own service directory. See Usage examples."
+
+      property :sv_timeout, Integer,
+        description: "Override the default sv timeout of 7 seconds."
+
+      property :sv_verbose, [TrueClass, FalseClass], default: false, description: "Whether to enable sv verbose mode. Default is false."
+
+      property :log_dir, String,
+        default: lazy { ::File.join("/var/log/", service_name) },
+        description: "The directory where the svlogd log service will run. Used when default_logger is true."
+
+      property :log_flags, String, default: "-tt",
+               description: "The flags to pass to the svlogd command. Used when default_logger is true."
+
+      property :log_size, Integer,
+        description: "The maximum size a log file can grow to before it is automatically rotated. See svlogd(8) for the default value."
+
+      property :log_num, Integer,
+        description: "The maximum number of log files that will be retained after rotation. See svlogd(8) for the default value."
```

Comment:
I've noticed these properties and their metadata components sometimes run to one to three lines long in this file. Will that have an effect on resource auto-generation?

Reply:
That'll be fine. It just makes it harder for humans to read here